### ISSUE TITLE
Model not learning

### ISSUE THREAD

**Issue by minaremeli** [minaremeli/MOON#12]
Running the model with the following parameters does not learn even after 100 rounds of training (this code ran for a total of 1.5 hours on `dev-gpu-0.cl.cam.ac.uk` ❗ I think this is less than expected).

```
python main.py --num_rounds 100 --sample_fraction_fit 1.0 --num_clients 10 --strategy fedAvg --beta 5
```

Very small changes (`1e-3` and smaller) can be observed over time on the loss (`CrossEntropyLoss`), but accuracy almost always stays a constant `0.1`. This is true for both local and global accuracies. I experienced the same when running for fewer rounds / different beta / num_clients / sample_fraction_fit / etc.

**Possible causes**:
- learning rate is too small
- weights don't get updated
- gradient vanishing (?)

I might need some help figuring this out @VasundharaAgarwal 🙏
